rpmlint/checks/FileDigestCheck.py

```diff
@@ -62,52 +65,45 @@ def _get_digest_configuration_group(self, pkgfile):
                     pass
         return None
 
-    def _check_filetypes(self, pkg):
-        """
-        Check that all symlinks point to a correct location and that
-        symlinks are allowed in a configuration group.
-        """
-        result = True
-        for filename, pkgfile in pkg.files.items():
-            group = self._get_digest_configuration_group(pkgfile)
-            if not group:
-                continue
-
-            if filename in pkg.ghost_files:
-                self.output.add_info('E', pkg, f'{group}-file-digest-ghost', filename)
-                result = False
-                continue
-
-            if stat.S_ISLNK(pkgfile.mode) and not self.follow_symlinks_in_group[group]:
-                self.output.add_info('E', pkg, f'{group}-file-symlink', filename)
-                result = False
-                continue
-        return result
-
-    def _is_valid_digest(self, pkgfile, digest, pkg):
+    def _is_valid_digest(self, path, digest, pkg):
         algorithm = digest['algorithm']
         if algorithm == 'skip':
             return (True, None)
 
+        pkgfile = self._resolveLinks(pkg, path)
+        if pkgfile is None:
+            return (False, None)
+
+        file_digest = self._calcDigest(pkgfile, algorithm)
+        return (file_digest == digest['hash'], file_digest)
+
+    def _resolveLinks(self, pkg, path):
```

Comment:
Please use standard Python naming for identifiers: `_resolve_links`.

Reply:
Thanks for the review. I rebased and force-pushed, the function names in code and also the commit messages are adjusted now.